src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,399 @@
+        {
+            get => ImmutableArray.Create(CS0266, CS1503);
+        }
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var diagnostic = context.Diagnostics.First();
+
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (root != null && syntaxFacts != null && semanticModel != null &&
+                TryGetTargetNode(root, diagnostic.Location.SourceSpan) is ExpressionSyntax targetNode)
+            {
+                var exactSolution = GetTypeInfo(semanticModel, root, targetNode, cancellationToken, out var nodeType, out var conversionType, out var potentialConvTypes);
+                if (exactSolution)
+                {
+                    context.RegisterCodeFix(new MyCodeAction(
+                        CSharpFeaturesResources.Add_explicit_cast,
+                        c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                        context.Diagnostics);
+                }
+                else if (potentialConvTypes.Length > 1)
+                {
+                    var actions = new ArrayBuilder<CodeAction>();
+                    for (var i = 0; i < Math.Min(MaximumConversionOptions, potentialConvTypes.Length); i++)
+                    {
+                        var convType = potentialConvTypes[i];
+                        actions.Add(new MyCodeAction(string.Format(CSharpFeaturesResources.Convert_type_to_0, convType.ToDisplayString()),
+                            c => FixIt(context.Document, root, targetNode, convType, c)));
+                    }
+
+                    if (potentialConvTypes.Length > 3)
+                    {
+                        Logger.Log(FunctionId.CodeFixes_AddExplicitCast,
+                            KeyValueLogMessage.Create(m =>
+                            {
+                                m["NumberOfCandidates"] = potentialConvTypes.Length;
+                            }));
+                    }
+
+                    context.RegisterCodeFix(new CodeAction.CodeActionWithNestedActions(
+                        CSharpFeaturesResources.Add_explicit_cast,
+                        actions.ToImmutableArray(), false),
+                        context.Diagnostics);
+                }
+            }
+        }
+
+        private async Task<Document> FixIt(Document document, SyntaxNode currentRoot, ExpressionSyntax targetNode, ITypeSymbol conversionType, CancellationToken cancellationToken)
```

Comment:
Needs a better name, say `ApplyFixAsync`. The method can also be a static method.